jl4-core/src/L4/Evaluate/ValueLazy.hs

```diff
@@ -78,7 +91,8 @@ instance NFData a => NFData (Value a) where
   rnf ValNil                      = ()
   rnf (ValCons r1 r2)             = rnf r1 `seq` rnf r2
   rnf (ValClosure given expr env) = env `seq` rnf given `seq` rnf expr
-  rnf (ValUnaryBuiltinFun r)      = rnf r
+  rnf (ValUnaryBuiltinFun r)      = r `seq` ()
+  rnf (ValBuiltinFun r)           = r `seq` ()
```

Comment:
same as above, i.e. ``r `seq` ()`` relies on `r` being in normalform after a simple `seq`. that doesn't have to be the case - future refactorings may break us without noticing.